### PR TITLE
DB-8156 Fix TRIM() trim type same as alias

### PR DESCRIPTION
**Pull request by changli6**

### CODE REVIEW

**Comment by changli6** [comment]
jenkins please test branch @cdh5.12.2

**Comment by spliceci** [comment]
TEST FAILED

**Comment by changli6** [comment]
jenkins please test branch @cdh5.12.2

**Comment by spliceci** [comment]
TEST SUCCEEDED +1

**Comment by carolp-503** [comment]
OK for R2.7 - Customer waiting

File: splice_machine/src/test/java/com/splicemachine/derby/impl/sql/execute/operations/TernaryOperationIT.java [hunk 1]
```diff
@@ -213,9 +213,51 @@ public void testTRIM() throws Exception {
...
+        rs = methodWatcher.executeQuery(sql);
+        rs.next();
+        iCell = rs.getInt(1);
+        Assert.assertEquals("Wrong trim result", 4, iCell);
+
+        sql = "select count(*) from ( select trim (t.c) from a t ) tab";
+        rs = methodWatcher.executeQuery(sql);
+        rs.next();
+        iCell = rs.getInt(1);
+        Assert.assertEquals("Wrong trim result", 4, iCell);
```
Comment (yxia92) [inline splice_machine/src/test/java/com/splicemachine/derby/impl/sql/execute/operations/TernaryOperationIT.java]: Can you add the following two test cases too:
1) negative test case:
select trim (leading.c) from a as leading;
2) positive test case:
select trim ("leading".c) from a as "leading";

**Comment by changli6** [comment]
The STRIP function supports the following formats:

STRIP(trimSource, trimType, trimChar)
STRIP(trimSource, trimType)

The LOOKAHEAD(<COMMA>) in line 7652 is used to identify if the user specified a trimChar. It consumes <COMMA> after the trimType.
The ansiTrimSpecFollows() consumed <COMMA> is the one before the trimType.

File: splice_machine/src/test/java/com/splicemachine/derby/impl/sql/execute/operations/TernaryOperationIT.java (hunk 1 above)
Comment (changli6) [reply splice_machine/src/test/java/com/splicemachine/derby/impl/sql/execute/operations/TernaryOperationIT.java]: sure, added these tests.

Comment (yxia92) [reply splice_machine/src/test/java/com/splicemachine/derby/impl/sql/execute/operations/TernaryOperationIT.java]: Thanks @changli6

**Comment by yxia92** [comment]
For the two scenarios of STRIP(without and without trimChar) that you listed, there is no ambiguity, and no need of LOOKAHEAD to determine which branch to go. That is why I asked to remove. If we look at the other path that does not contains TrimSpec, it also need to handle two scenarios:

STRIP(trimSource, trimChar)
STRIP(trimSource)

and no LOOKAHEAD is used.

```
        LOOKAHEAD ({!ansiTrimSpecFollows()})
        (
            // <source> COMMA <trimChar>
            <COMMA> trimChar = additiveExpression(null,0) <RIGHT_PAREN>
            {
                return getTrimOperatorNode(trimSpec, trimChar, trimSource, null);
            }
        |
            // <source>
            <RIGHT_PAREN>
            {
                return getTrimOperatorNode(trimSpec, null, trimSource, null);
            }

        )
```

**Comment by changli6** [comment]
I got your point @yxia92 It is a useless line. Thank you!

**Pull request merged**